hyperparameter_tuning.py

```diff
@@ -0,0 +1,155 @@
+import gc
+
+import numpy as np
+from deepsense import neptune
+from sklearn.externals import joblib
+
+from steps.base import BaseTransformer
+from utils import set_seed
+
+
+class RandomSearchOptimizer(BaseTransformer):
+    def __init__(self, TransformerClass, params,
+                 score_func, maximize,
+                 train_input_keys, valid_input_keys,
+                 n_runs,
+                 callbacks=[]):
+        self.TransformerClass = TransformerClass
```

Comment:
Here, it seems that you inject transformer inside the transformer. I think we should refactor this in a way that it does not requires this trick